core/src/main/scala/kafka/coordinator/transaction/TransactionLog.scala

```diff
@@ -70,17 +71,18 @@ object TransactionLog {
     def ofVersion(version: Int): Option[Schema] = SCHEMAS.get(version)
   }
 
-  private object ValueSchema {
-    private val ProducerIdKey = "producer_id"
-    private val ProducerEpochKey = "producer_epoch"
-    private val TxnTimeoutKey = "transaction_timeout"
-    private val TxnStatusKey = "transaction_status"
-    private val TxnPartitionsKey = "transaction_partitions"
-    private val TxnEntryTimestampKey = "transaction_entry_timestamp"
-    private val TxnStartTimestampKey = "transaction_start_timestamp"
-
-    private val PartitionIdsKey = "partition_ids"
-    private val TopicKey = "topic"
+  private[coordinator] object ValueSchema {
+    val ProducerIdKey = "producer_id"
+    val ProducerEpochKey = "producer_epoch"
+    val LastProducerEpochKey = "last_producer_epoch"
```

Comment:
To handle the wraparound case, it seems like we might have to store last_producer_id as well?